core/trino-main/src/main/java/io/trino/execution/resourcegroups/InternalResourceGroupManager.java

```diff
@@ -118,7 +118,7 @@ public void submit(ManagedQueryExecution queryExecution, SelectionContext<C> sel
     public SelectionContext<C> selectGroup(SelectionCriteria criteria)
     {
         return configurationManager.get().match(criteria)
-                .orElseThrow(() -> new TrinoException(QUERY_REJECTED, "Query did not match any selection rule"));
+                .orElseThrow(() -> new TrinoException(QUERY_REJECTED, "User does not belong to any enabled resource group"));
```

Comment:
Maybe

No matching resource group found with configured selection rules. No resources for query processing available.

Or only part of that even.

Reply:
sounds good to me

Reply:
@mosabua please review it.